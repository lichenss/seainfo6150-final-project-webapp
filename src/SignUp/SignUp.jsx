import React from 'react'
import Footer from '../Footer/Footer'
import Header from '../Header/Header'
import styles from './SignUp.module.css'
import { useHistory } from 'react-router-dom'

const SignUp = () => {
    const history = useHistory()
    const handleSubmit=()=>{history.push('/thank-you')}
    const handleCancel=()=>{history.push('/')}
    return (
        <div>
            <Header />
            <section className={styles.section}>
            <div className={styles.container}>  
            <div className={styles.titleContainer}>
            <h1 className={styles.title}>Sign Up</h1>
            </div>
            <div className={styles.formContainer}>
            <form className={styles.form} onSubmit={handleSubmit} >
                <div>
                <label className={styles.label}>Username</label>
                <input className={styles.input}
                type="text" required
                />
                </div>
                <div>
                <label className={styles.label}>Name</label>
                <input className={styles.input}
                type="text" required
                />
                </div>
                <div>
                <label className={styles.label}>Email</label>
                <input className={styles.input}
                type="email" placeholder="xxx@xxx.com" 
                required
                />
                </div>
                <div>
                <label className={styles.label}>Password</label>
                <input className={styles.input}
                type="password" required
                />
                </div>
                <div>
                <label className={styles.label}>Phone Number</label>
                <input className={styles.input} type="tel" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" placeholder="xxx-xxx-xxxx"
                 required
                />
                </div>
                <div>
                 <div>
                 <label for="gender" className={styles.label}>Gender</label>
                 <select id="gender" name="gender">
                     <option value="male">Male</option>
                     <option value="female">Female</option>
                     <option value="other">Other</option>
                     </select>   
                </div>
                <div>
                 <label for="degree" className={styles.label}>Higest Degree</label>
                 <select id="degree" name="degree">
                     <option value="bachelor">Bachelor Degree Or Above</option>
                     <option value="hign-school">High School</option>
                     <option value="other">Other</option>
                     </select>
                <div>
                <p>Select a your age:</p>
                <input type="radio" id="r1" name="r1" value="r1" checked />
                <label for="r1">Under 18</label>
                <input type="radio" id="r2" name="r1" value="r2" checked/>
                <label for="r2">18-40</label>
                <input type="radio" id="r3" name="r1" value="r3" checked/>
                <label for="r3">Above 40</label>
                </div>    
                </div>
                </div>
                <div className={styles.btnContainer}>
                <button className={styles.button} onClick={handleCancel}> Cancel</button>
                <button className={styles.button} type="submit"> Sign Up</button>
                </div>
            </form>
            </div>
            </div>
            </section>
            <Footer />
        </div>
    )
}
export default SignUp