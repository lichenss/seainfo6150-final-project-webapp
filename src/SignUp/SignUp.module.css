.section {
    padding-top: 70px;
    display: flex;
    flex-direction: column;
    background-color: rgb(229, 231, 233);
    padding-bottom: 3rem;
}
.form{
    display: flex;
    flex-direction: column;
    place-items: left;
    padding-bottom: 3rem;
}
.label{
    font-family: 'Roboto', sans-serif;
}
.button{
    margin-left: 10px;
    margin-right: 10px;
}