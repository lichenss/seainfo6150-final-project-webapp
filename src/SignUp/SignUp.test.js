import React from "react";
import SignUp from "./SignUp";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
describe("SignUp tests", () => {
    it("renders correctly", () => {
      const { container } = render(
      <BrowserRouter>
      <SignUp />
      </BrowserRouter>);
      expect(container).toMatchSnapshot();
    });
  });