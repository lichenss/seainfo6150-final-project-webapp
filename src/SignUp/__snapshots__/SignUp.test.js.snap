// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp tests renders correctly 1`] = `
<div>
  <div>
    <div>
      <header
        class="header"
      >
        <div
          class="navbar"
        >
          <button
            class="logo"
          />
          <div>
            <a
              class="catLink"
              href="/category/computer-science"
            >
              Computer Science
            </a>
            <a
              class="catLink"
              href="/category/business"
            >
              Business
            </a>
            <a
              class="catLink"
              href="/category/data-science"
            >
              Data Science
            </a>
            <a
              class="catLink"
              href="/category/arts-and-humanities"
            >
              Arts and Humanities
            </a>
            <a
              class="catLink"
              href="/category/physical-science-and-engineering"
            >
              Physical Science and Engineering
            </a>
            <a
              class="catLink"
              href="/category/math-and-logic"
            >
              Math and Logic
            </a>
          </div>
          <div
            class="links"
          >
            <a
              class="barLink"
              href="/log-in"
            >
              Log In
            </a>
            <a
              class="barLink"
              href="/sign-up"
            >
              Sign Up
            </a>
          </div>
        </div>
      </header>
    </div>
    <section
      class="section"
    >
      <div
        class="container"
      >
        <div
          class="titleContainer"
        >
          <h1
            class="title"
          >
            Sign Up
          </h1>
        </div>
        <div
          class="formContainer"
        >
          <form
            class="form"
          >
            <div>
              <label
                class="label"
              >
                Username
              </label>
              <input
                class="input"
                required=""
                type="text"
              />
            </div>
            <div>
              <label
                class="label"
              >
                Name
              </label>
              <input
                class="input"
                required=""
                type="text"
              />
            </div>
            <div>
              <label
                class="label"
              >
                Email
              </label>
              <input
                class="input"
                placeholder="xxx@xxx.com"
                required=""
                type="email"
              />
            </div>
            <div>
              <label
                class="label"
              >
                Password
              </label>
              <input
                class="input"
                required=""
                type="password"
              />
            </div>
            <div>
              <label
                class="label"
              >
                Phone Number
              </label>
              <input
                class="input"
                pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
                placeholder="xxx-xxx-xxxx"
                required=""
                type="tel"
              />
            </div>
            <div>
              <div>
                <label
                  class="label"
                  for="gender"
                >
                  Gender
                </label>
                <select
                  id="gender"
                  name="gender"
                >
                  <option
                    value="male"
                  >
                    Male
                  </option>
                  <option
                    value="female"
                  >
                    Female
                  </option>
                  <option
                    value="other"
                  >
                    Other
                  </option>
                </select>
              </div>
              <div>
                <label
                  class="label"
                  for="degree"
                >
                  Higest Degree
                </label>
                <select
                  id="degree"
                  name="degree"
                >
                  <option
                    value="bachelor"
                  >
                    Bachelor Degree Or Above
                  </option>
                  <option
                    value="hign-school"
                  >
                    High School
                  </option>
                  <option
                    value="other"
                  >
                    Other
                  </option>
                </select>
                <div>
                  <p>
                    Select a your age:
                  </p>
                  <input
                    checked=""
                    id="r1"
                    name="r1"
                    type="radio"
                    value="r1"
                  />
                  <label
                    for="r1"
                  >
                    Under 18
                  </label>
                  <input
                    checked=""
                    id="r2"
                    name="r1"
                    type="radio"
                    value="r2"
                  />
                  <label
                    for="r2"
                  >
                    18-40
                  </label>
                  <input
                    checked=""
                    id="r3"
                    name="r1"
                    type="radio"
                    value="r3"
                  />
                  <label
                    for="r3"
                  >
                    Above 40
                  </label>
                </div>
              </div>
            </div>
            <div
              class="btnContainer"
            >
              <button
                class="button"
              >
                 Cancel
              </button>
              <button
                class="button"
                type="submit"
              >
                 Sign Up
              </button>
            </div>
          </form>
        </div>
      </div>
    </section>
    <footer
      class="footer"
    >
      <div
        class="links"
      >
        <a
          class="link"
          href="/contact-us"
        >
          <h3>
            Contact Us
          </h3>
        </a>
        <a
          class="link"
          href="/about-us"
        >
          <h3>
            About Us
          </h3>
        </a>
      </div>
    </footer>
  </div>
</div>
`;
